packages/main/src/TableVirtualizer.ts

```diff
@@ -0,0 +1,279 @@
+/* eslint-disable no-bitwise */
+import {
+	isUp,
+	isUpShift,
+	isDown,
+	isDownShift,
+	isPageUp,
+	isPageDown,
+	isHome,
+	isEnd,
+	isTabNext,
+	isTabPrevious,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import UI5Element, { type InvalidationInfo } from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import getActiveElement from "@ui5/webcomponents-base/dist/util/getActiveElement.js";
+import { getTabbableElements } from "@ui5/webcomponents-base/dist/util/TabbableElements.js";
+import { throttle } from "./TableUtils.js";
+import type { ITableFeature } from "./Table.js";
+import type Table from "./Table.js";
+
+enum TabBlocking {
+	None = 0,
+	Next = 1,
+	Previous = 2,
+	Released = 4,
+}
+
+/**
+ * Fired when the virtualizer is changed by user interaction e.g. on scrolling.
+ * @param {first} number The 0-based index of the first children currently rendered
+ * @param {last} number The 0-based index of the last children currently rendered
+ * @public
+ */
+type RangeChangeEventDetail = {
+	first: number,
+	last: number,
+};
+
+/**
+ * @class
```

Comment:
It breaks if `overflowMode=Popin`. Maybe a validation would be nice here, that either throws an error or "corrects" the value to `Scroll`
Should be documented as well